src/plone/restapi/services/discussion/conversation.py

```diff
@@ -38,7 +38,7 @@ class CommentsGet(Service):
 
     def publishTraverse(self, request, name):
         if name:
-            self.comment_id = int(name)
+            self.comment_id = parse_int(dict(comment_id=name), "comment_id", None)
```

Comment:
This is okay, but a bit awkward. Maybe the first parameter of parse_int should be the actual value rather than a dict which contains the value? Not a deal breaker though.

Reply:
then you need to repeat the prop anyway to get a nicer error message in the exception...
or have parse_int and parse_int_dict.

Reply:
Yeah. It can stay like this.

Reply:
or I can do this. Just a little less explicit

```
def parse_int(data, prop, default):
    """
    Args:
        data: dict from a request to get str, or str itself
        prop: name of a integer paramater in the dict
        default: default if not found

    Returns: an integer
    Raises: BadRequest if not an int
    """
    value = data if type(data) in (str, int) else data.get(prop, default)
    try:
        return int(value)
    except (ValueError, TypeError):
        raise BadRequest(f"Invalid {prop}: Not an integer")
```